plugins/providers/maxcompute/provider.go

```diff
@@ -130,26 +131,60 @@ func (p *provider) GetResources(ctx context.Context, pc *domain.ProviderConfig)
 	}
 
 	if slices.Contains(availableResourceTypes, resourceTypeTable) {
-		tableRes, err := client.ListTables(project.Name, &maxcompute.ListTablesRequest{})
+		odpsClient, err := p.getOdpsClient(pc, "")
 		if err != nil {
-			return nil, fmt.Errorf("failed to list tables for project %s: %w", *project.Name, err)
+			return nil, err
+		}
+
+		schemaRes := map[string]struct{}{"default": {}}
+		err = odpsClient.Project(bptr.ToStringSafe(project.Name)).Schemas().List(func(schema *odps.Schema, err error) {
+			if schema == nil {
+				return
+			}
+			if slices.Contains(credentials.ExcludedSchemas, schema.Name()) {
+				return
+			}
+			schemaRes[schema.Name()] = struct{}{}
+		})
+		if err != nil {
+			return nil, err
 		}
 
-		for _, table := range tableRes.Body.Data.Tables {
-			var urn string
-			if table.Schema == nil {
-				urn = fmt.Sprintf("%s.%s", *project.Name, *table.Name)
-			} else {
-				urn = fmt.Sprintf("%s.%s.%s", *project.Name, *table.Schema, *table.Name)
+		for schemaName := range schemaRes {
+			var marker *string
+			var tableRes []*maxcompute.ListTablesResponseBodyDataTables
+			for {
+				tmpTableRes, err := client.ListTables(project.Name, &maxcompute.ListTablesRequest{
+					Marker:     marker,
+					SchemaName: bptr.FromString(schemaName),
+				})
+				if err != nil {
+					return nil, fmt.Errorf("failed to list tables for project %s: %w", *project.Name, err)
```

Comment:
let's make it more verbose by also specifying the schema name in the error
```suggestion
					return nil, fmt.Errorf("failed to list tables for schema %q: %w", *project.Name + "." + schemaName, err)
```